kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -732,7 +732,8 @@ unifyEqualsNormalized
     renormalized <- normalize1 normalizedTerm
 
     let unifierTerm :: TermLike variable
-        unifierTerm = asInternal tools sort1 renormalized
+        unifierTerm =
+            TermLike.markSimplified $ asInternal tools sort1 renormalized
```

Comment:
Shouldn't this also test that the keys/values/opaque stuff are simplified? Either as an assertion (e.g. `TermLike.markSimplified` could assert that all direct children are simplified), or as an actual computation input. Similar question for some of the other uses of `markSimplified`.

Reply:
No, we should not be inspecting children here. It is not necessary because the simplifier recurses through the entire pattern, so if the children were left unsimplified, an error would already have been thrown. It would also cause a lot of spurious test failures because all the unit tests send un-simplified patterns.

Reply:
This is also called outside of the simplifier, i.e. by the unifier, which should, in principle, also sends simplified stuff, but that might not be always so. I mean, do we actually simplify the left hand term of rules? They are probably simplified since we internalize stuff, and usually there's nothing to simplify there, but I don't think that would be true if we had a rule with a function symbol in the left.

So, if my theory above is valid, then it's not always obvious that we are using `TermLike.markSimplified` properly, so maybe it would be better to do the check and find a reasonable way to fix those tests. If it's not valid, then I'll just approve the PR.

Reply:
Discussed on Slack, this `markSimplified` should go away.

Reply:
Sorry, I think I misunderstood what you were saying. `unifyEqualsNormalized` is the "canonical" simplifier for Map and Set, so this should use `markSimplified`, but you are right that it should _only_ do that if the children are simplified.